### PR TITLE
feat(web): Duolingo-style L1–L5 lesson pathway in HomeView

### PR DESCRIPTION
**Pull request by auerbachb**
## **User description**

Closes #336

## What

Adds a Duolingo-style lesson pathway to the web `HomeView`, embedded between the day display and the action buttons. Lesson nodes are grouped into five levels (L1 Stillness, L2 Awareness, L3 Focus, L4 Intention, L5 Kindness), each with 10 day-nodes.

Node states (derived purely from `currentDay`, no backend change):
- **completed** (green check) — `day < currentDay`
- **current** (amber, animated glow) — `day === currentDay`
- **locked** (dimmed, visible) — `day > currentDay`

v1 unlock rule: completing a standard sit advances `currentDay`, which unlocks the next node.

## Walkthrough

[pathway_unlock_progression_demo.mp4](https://cursor.com/agents/bc-728cadb0-9f1a-4725-8bf2-570273330b2e/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fpathway_unlock_progression_demo.mp4)

Unlock progression: the amber current node advances and passed nodes turn green; `set 55` shows the fully-completed pathway.

[Pathway at day 13](https://cursor.com/agents/bc-728cadb0-9f1a-4725-8bf2-570273330b2e/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fpathway_day13.webp)
[Pathway at day 1](https://cursor.com/agents/bc-728cadb0-9f1a-4725-8bf2-570273330b2e/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fpathway_day1.webp)
[Pathway fully completed](https://cursor.com/agents/bc-728cadb0-9f1a-4725-8bf2-570273330b2e/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fpathway_day55_complete.webp)

## Review feedback addressed

- **Connector into current node** (Bugbot): the path line now greens based on the *preceding* node's completed state, so it reaches into the amber current node instead of stopping one step short.<br />[Green connector into current node](https://cursor.com/agents/bc-728cadb0-9f1a-4725-8bf2-570273330b2e/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fpathway_fix_connector_green.webp)
- **Narrow-viewport overflow** (Bugbot): nodes are now shrink-to-fit (`flex: 0 1 30px` + `aspect-ratio: 1`), so all 10 nodes fit without clipping on small phones.<br />[All nodes fit at 360px width](https://cursor.com/agents/bc-728cadb0-9f1a-4725-8bf2-570273330b2e/artifacts?path=%2Fopt%2Fcursor%2Fartifacts%2Fpathway_fix_narrow_360px.webp)
- **Keyframe naming** (CodeRabbit): renamed `pathwayPulse` → `pathway-pulse` for stylelint kebab-case.
- **Auto-scroll to current node** (CodeRabbit): intentionally not implemented — it is outside the locked CR plan / issue ACs, and a window-level `scrollIntoView` on this inline section would jump the page past the brand/day header every time Home opens. Can be revisited as a follow-up if desired.

## Files
- `src/lib/pathway.ts` — pure level/node derivation (`buildPathway`, `nodeStateForDay`).
- `src/lib/pathway.test.ts` — unit tests for derivation + edge cases.
- `src/components/Pathway.tsx` — pathway UI (CSS vars, inline styles, JetBrains Mono).
- `src/components/HomeView.tsx` — embeds `<Pathway />` between day display and CTAs.
- `src/app/globals.css` — `pathway-pulse` keyframe for the current node (respects `prefers-reduced-motion`).

Web-only; no migration, no API change. iOS parity is a separate follow-up.

## Test plan
- ✅ `npm run test:unit` (232 passed, incl. new `pathway.test.ts`)
- ✅ `npm run typecheck`
- ✅ Manual: verified completed/current/locked rendering, unlock progression, the green connector fix, and narrow-viewport (360px) fit (artifacts above).

<sub>To show artifacts inline, <a href="https://cursor.com/dashboard/cloud-agents#my-pull-requests">enable</a> in settings.</sub>

<div><a href="https://cursor.com/agents/bc-728cadb0-9f1a-4725-8bf2-570273330b2e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-728cadb0-9f1a-4725-8bf2-570273330b2e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

## Summary by CodeRabbit

* **New Features**
  * Added a “Lesson pathway” section to the home screen with level/day progress.
  * Steps now reflect completed/current/locked states, including a pulsing highlight for the current step.
* **Bug Fixes**
  * Improved pathway progress handling for edge cases, including invalid, non-finite, or out-of-range day values (with correct current-step behavior at the max day).
* **Tests**
  * Added automated coverage for pathway state transitions, level grouping, and boundary conditions.

___

## **CodeAnt-AI Description**
Add a Duolingo-style lesson pathway to the Home screen

### What Changed
- Home now shows a five-level lesson pathway between the day display and the main action button, with each day marked as completed, current, or locked
- Passed days show a green check, the current day is highlighted with an amber glow, and future days stay visible but dimmed
- The pathway now stays within its own scroll area on small screens so the main button does not get pushed into the bottom nav
- The pathway can be focused and scrolled with the keyboard, and the active focus state is visible
- The connector line now reaches into the current lesson instead of stopping one step short

### Impact
`✅ Clearer lesson progress`
`✅ Fewer mobile layout overlaps`
`✅ Keyboard-accessible lesson navigation`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

**Comment by auerbachb** [comment]
@graphite-app re-review

File: src/components/Pathway.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                  <div
+                    aria-hidden="true"
+                    style={{
+                      flex: 1,
+                      height: "1px",
+                      minWidth: "4px",
+                      background:
+                        node.state === "completed"
+                          ? "var(--accent-green-border)"
+                          : "var(--border-1)",
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply src/components/Pathway.tsx:173]: Fixed: the connector uses `level.nodes[i - 1]?.state === 'completed'` (Pathway.tsx line ~163) so the path segment is green when the *preceding* node is finished, meaning the green line reaches all the way into the amber current node rather than stopping one step short. This was already in place in commit 8076ede; acknowledging now.

File: src/components/Pathway.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                    }}
+                  />
+                )}
+                <Node node={node} />
+              </div>
+            ))}
+          </div>
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply src/components/Pathway.tsx:180]: Fixed in two parts: (1) nodes use `flex: '0 1 30px'` (grow:0, shrink:1, basis:30px) with `minWidth:0` and `aspectRatio:'1/1'` so they shrink-to-fit on narrow phones — already present in commit 8076ede. (2) The section now has `maxHeight: clamp(160px, 30vh, 240px)` with `overflowY: auto` (commit 2a54500) so the pathway scrolls within its container instead of pushing the Begin CTA into the fixed bottom nav — this also resolves the mobile safe-area e2e CI failure.

**Comment by auerbachb** [comment]
@codeant-ai review

**Comment by auerbachb** [comment]
Fixed in bb1f4f9: added tabIndex={0} to the overflow <section> so keyboard-only users can focus it and scroll lower levels into view with arrow keys / Page Down. A :focus-visible rule in globals.css provides an amber focus ring so the focused state is visually obvious. The browser default outline is suppressed in inline styles to avoid a double ring.

**Comment by auerbachb** [comment]
@codeant-ai review

File: src/components/Pathway.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        // the Begin CTA into the fixed bottom nav on narrow/short viewports
+        // (e.g. iPhone SE 375×667). clamp: 160px floor, 30vh mid, 240px ceiling.
+        maxHeight: "clamp(160px, 30vh, 240px)",
+        overflowY: "auto",
+        display: "flex",
+        flexDirection: "column",
+        gap: "var(--s4)",
+        // focus-visible outline supplied via :focus-visible in globals.css;
+        // suppress the default browser outline here to avoid double ring.
+        outline: "none",
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply src/components/Pathway.tsx]: Fixed in 686a1ee: removed the inline `outline: "none"` and its comment from the Pathway `<section>` style. The `:focus-visible` selector in globals.css now takes effect correctly — keyboard users will see the amber focus ring. No double-ring issue since browsers suppress the default outline for pointer interactions when a `:focus-visible` rule is present.

**Comment by auerbachb** [comment]
@codeant-ai review
